File: HW/HW9/TreeNode.java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;

    }
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q)
    {
        int parentValue = root.val;
        int pValue = p.val;
        int qValue = q.val;

        if (pValue > parentValue && qValue > parentValue)
        {
            return lowestCommonAncestor(root.right, p, q);
        }
        else if (pValue < parentValue && qValue < parentValue)
        {
            return lowestCommonAncestor(root.left, p, q);
        }
        else
        {
            return root;
        }
    }
}

File: Labs/Lab 4/TreeNode.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

    def invertTree(self, root):
        if root is None:
            return None

        root.left, root.right = self.invertTree(root.right), self.invertTree(root.left)
        return root
